templates/core/terraform/keyvault/keyvault.tf

```diff
@@ -0,0 +1,39 @@
+resource "azurerm_key_vault" "kv" {
+  name                = "kv-${var.resource_name_prefix}-${var.environment}-${var.tre_id}"
+  location            = var.location
+  resource_group_name = var.resource_group_name
+  sku_name            = "standard"
+  purge_protection_enabled = true
+  tenant_id           = var.tenant_id
+}
+
+resource "azurerm_private_dns_zone" "vaultcore" {
+  name                = "privatelink.vaultcore.azure.net"
+  resource_group_name = var.resource_group_name
+}
+
+resource "azurerm_private_dns_zone_virtual_network_link" "vaultcorelink" {
+  name                  = "vaultcorelink"
+  resource_group_name   = var.resource_group_name
+  private_dns_zone_name = azurerm_private_dns_zone.vaultcore.name
+  virtual_network_id    = var.core_vnet
+}
+
+resource "azurerm_private_endpoint" "kvpe" {
+  name                = "pe-kv-${var.resource_name_prefix}-${var.environment}-${var.tre_id}"
+  location            = var.location
+  resource_group_name = var.resource_group_name
+  subnet_id           = var.shared_subnet
```

Comment:
Is the shared subnet appropriate? Is the key vault really shared by workspace users, or is it actually just used by composition service and for certs?

Reply:
Valid question, everything with PE is right now hookedup into the Shared. I would run with this as a initial, and refine as we go along.

Reply:
@christoferlof can we have an issue for this. Given private endpoints don't obey nsg rules at present, I tend to keep them separate, and need to ensure traffic is blocked outbound to that subnet from everything else.

Also what's the NSG setup in this PR? Need another issue for that?

Reply:
Please see #53